app/assets/javascripts/uploadFiles.js

```diff
@@ -0,0 +1,222 @@
+;(function (global) {
+
+    var fileList = (function () {
+
+        var _files = [];
+        var _tableId;
+
+        var readableSize = function formatBytes(bytes) {
+            var decimals = 2;
+            if (bytes === 0) return '0 Bytes';
+            var k = 1024,
+                dm = decimals <= 0 ? 0 : decimals || 2,
+                sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'],
+                i = Math.floor(Math.log(bytes) / Math.log(k));
+            return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+        };
```

Comment:
is `;` needed here?

Reply:
A function declaration does not need (and should not have) a semicolon following it:
`
function test(o) {
}
` 
However, if you write a function as a statement, like the variable initializer below, then the statement should be terminated with a semicolon, just like any other statement would be:
` 
var a = function test(o) {
};
`